bids/layout/bids_validator.py

```diff
@@ -252,3 +256,151 @@ def conditional_match(self, expression, path):
+    >>> dfs = validate_sequences(layout, 'pybids/bids/layout/tests/data/sample_validation_config.json')
+    >>> dfs.duplicates
+    # Put example output here
+    >>> df.summary
+    # Put example output here
+    >>> df.problems
+    # Put example output here
+    """
+    
+    duplicate_file_df = duplicate_check(layout)
+    summary_df, problem_df = expected_file_check(layout, config)
+    output = namedtuple('output', ['duplicates', 'summary', 'problems'])
+    return output(duplicate_file_df, summary_df, problem_df)
+    
+    
+def duplicate_check(layout):
+    """Checks images in BIDS project are not duplicated.
+
+    Check whether any files have duplicate content within the 
+    BIDS data set. Returns a data frame: duplicate_file_df.
+
+
+    Parameters
+    ----------
+        layout: BIDSLayout class
+            A BIDSLayout path of a data set.
+
+
+    Examples
+    --------
+    >>> layout = bids.grabbids.BIDSLayout('/path_to/sample_project_root')
+    >>> duplicate_file_df = duplicate_check(layout)
+    >>> duplicate_file_df
+    # Put example output here
+
+
+    Notes
+    ------
+    Returns a data frame in which the first column is the file
+    identifier and the second column is the path to the file.
+    Files with matching identifiers have the same content.
+    """
+    
+    def md5(fname):
+        hash_md5 = hashlib.md5()
+        with open(fname, "rb") as f:
+            for chunk in iter(lambda: f.read(4096), b""):
+                hash_md5.update(chunk)
+        return hash_md5.hexdigest()
+    hash_map = {}
+    all_niftis = layout.get(return_type="file", extensions='.nii.gz')
+    for nifti_file in all_niftis:
+        md5sum = md5(nifti_file)
+        if md5sum in hash_map:
+            hash_map[md5sum].append(nifti_file)
+        else:
+            hash_map[md5sum] = [nifti_file]
+    df = pd.DataFrame.from_dict(hash_map, orient='index') 
+    pruned_df = df.stack().reset_index().drop(columns='level_1')
+    out_df = pruned_df.rename(columns={'level_0': 'hash', 0: 'filename'})
+    return out_df
+    
+    
+def expected_file_check(layout, config):
+    """Checks files in BIDS project match user defined expectations.
+
+    This method checks the number of data set files against a user customized 
+    configuration file. Returns two data frames: summary_df, problem_df.
+
+
+    Parameters
+    ----------
+        layout: BIDSLayout class
+            A BIDSLayout path of a data set.
+
+        config: string
+            Path to customized configuration file.
+
+
+    Examples
+    --------
+    >>> layout = bids.grabbids.BIDSLayout('/path_to/sample_project_root')
+    >>> summary_df, problem_df = expected_file_check(layout, 'pybids/bids/layout/tests/data/sample_validation_config.json')
+    >>> summary_df
+    # Put example output here
+    >>> problem_df
+    # Put example output here
+
+
+    Notes
+    --------
+    The configuration file must contain the following keys: 
```

Comment:
It doesn't look like these are actually mandatory, as they're just passed on to the `Layout.get()` query, and will work fine even if some are empty. E.g., many datasets don't have more than one session, so there's no reason to explicitly pass session.